code/modules/supply/supply_console.dm

```diff
@@ -0,0 +1,499 @@
+			if(!ordernum)
+				return
+			deny_crate(user, ordernum)
+
+
+		// Popup to show CC message logs. Its easier this way to avoid box-spam in TGUI
+		if("showMessages")
+			// Public consoles cant view messages
+			if(is_public)
+				return
+			var/datum/browser/ccmsg_browser = new(user, "ccmsg", "Central Command Cargo Message Log", 800, 600)
+			ccmsg_browser.set_content(SSeconomy.centcom_message)
+			ccmsg_browser.open()
+
+/obj/machinery/computer/supplycomp/proc/order_crate(mob/user, datum/supply_order/order, datum/money_account/account)
+	var/datum/money_account/selected_account = account
+	if(!istype(selected_account)) //if no account found, just default to cargo account
+		CRASH("order_crate called without a specified money account")
+	else
+		if(selected_account.account_type == ACCOUNT_TYPE_DEPARTMENT)
+			for(var/datum/station_department/department in SSjobs.station_departments)
+				if(department.department_account == selected_account)
+					order.ordered_by_department = department //now that we know which department this is for, attach it to the order
+					order.orderedbyaccount = selected_account
+					order.requires_head_approval = TRUE
+
+					if(length(order.object.department_restrictions))
+						//this crate has a department whitelist description
+						if(!(department.department_name in order.object.department_restrictions))
+							//this department is not in this whitelist, require QM approval
+							order.requires_qm_approval = TRUE
+					break
+		else if(selected_account.account_type == ACCOUNT_TYPE_PERSONAL && length(order.object.department_restrictions))
+			order.requires_qm_approval = TRUE
+
+	//===Handle Supply Order===
+	if(selected_account.account_type == ACCOUNT_TYPE_PERSONAL)
+		//if the account is a personal account (and doesn't require QM approval), go ahead and pay for it now
+		order.orderedbyaccount = selected_account
+		if(attempt_account_authentification(selected_account, user))
+			var/paid_for = FALSE
+			if(!order.requires_qm_approval && pay_with_account(selected_account, order.object.cost, "[order.object.name] Crate Purchase", "Cargo Requests Console", user, account_database.vendor_account))
+				paid_for = TRUE
+			SSeconomy.process_supply_order(order, paid_for) //add order to shopping list
+	else //if its a department account with pin or higher security or need QM approval, go ahead and add this to the departments section in request list
+		SSeconomy.process_supply_order(order, FALSE)
+		if(order.ordered_by_department.crate_auto_approve && order.ordered_by_department.auto_approval_cap >= order.object.cost)
+			approve_crate(user, order.ordernum)
+		investigate_log("| [key_name(user)] has placed an order for [order.object.amount] [order.object.name] with reason: '[order.comment]'", "cargo")
+
+/obj/machinery/computer/supplycomp/proc/approve_crate(mob/user, order_num)
+	for(var/datum/supply_order/department_order in SSeconomy.request_list)
+		if(department_order.ordernum == order_num)
```

Comment:
Can you make this into an early continue instead?

Reply:
done in next commit